### PR TITLE
hopskipjump attack support for empirical robustness metric

### PR DESCRIPTION
**Pull request by ggh96**
Compute empirical robustness score for the adversarial samples created by hopskipjump attack.

### CODE REVIEW

**Comment by beat-buesser** [comment]
Hi @ggh96 Thank you for this pull request. This pull request replaces #211.

**Comment by beat-buesser** [comment]
Fixes issue #198

File: art/metrics/metrics.py [hunk 1]
```diff
@@ -26,17 +26,25 @@
 
 import numpy as np
 import numpy.linalg as la
+import matplotlib.pyplot as plt
```
Comment (beat-buesser) [inline art/metrics/metrics.py]:
```suggestion
```

File: art/metrics/metrics.py [hunk 2]
```diff
@@ -26,17 +26,25 @@
...
 logger = logging.getLogger(__name__)
 
 SUPPORTED_METHODS = {
     "fgsm": {"class": FastGradientMethod, "params": {"eps_step": 0.1, "eps_max": 1., "clip_min": 0., "clip_max": 1.}},
+    "hsj":{"class": HopSkipJump, "params":{
+                  'max_iter': 50,
+                  'max_eval': 10000,
+                  'init_eval': 100,
+                  'init_size': 100,
+                  }}
```
Comment (beat-buesser) [inline art/metrics/metrics.py]: Could you please change this to 1 or 2 lines similar as for fgsm?

File: art/metrics/metrics.py [hunk 3]
```diff
@@ -77,13 +85,15 @@ def empirical_robustness(classifier, x, attack_name, attack_params=None):
...
-                        (Fast Gradient Sign Method)
+    :param attack_name: A string specifying the attack to be used. Currently supported attacks are {`fgsm', `hsj`}
+                        (Fast Gradient Sign Method, Hop Skip Jump)
     :type attack_name: `str`
     :param attack_params: A dictionary with attack-specific parameters. If the attack has a norm attribute, then it will
                           be used as the norm for calculating the robustness; otherwise the standard Euclidean distance
                           is used (norm=2).
     :type attack_params: `dict`
+    :param round: Round the features if they are one-hot encoded
+    :type round: 'bool'
```
Comment (beat-buesser) [inline art/metrics/metrics.py]: Is the documentation for `round` still required? If not, please remove it.

File: art/metrics/metrics.py (hunk 2 above)
Comment (ggh96) [reply art/metrics/metrics.py]: Done

File: art/metrics/metrics.py (hunk 3 above)
Comment (ggh96) [reply art/metrics/metrics.py]: Removed

**Comment by beat-buesser** [comment]
@ggh96 Thank you very much for your contribution!

**Pull request merged**